### ISSUE TITLE
`check-newsfragments` - pre-commit -  unclear error message when fragment doesn't end with `.`

### ISSUE THREAD

**Issue by orSolocate** [PyCQA/pylint#7593]
### Bug description

When `check-newsfragments` checker fails in `pre-commit` because of no `.` in the end of description the error message is not clear. should be fixed.``

Steps to reproduce:

1. Checkout to main branch.
2. run in terminal `towncreier create 7777.new_check`
3. open `7777.new_check` file and write the following:

```
Bla bla.
bla bla

Closes #7777
```

4. Stage `7777.new_check`
5. run `pre-commit` in terminal

Adding a `.` to the second at the end solves the pre-commit error, but the error message is unclear.

P.S the error message when running the `newsfragments` job in CI actually has a clearer error message that helps you understand what to do to fix the error.

### Command used

run `pre-commit` in terminal

### Pylint output

result in log:
```
Check newsfragments......................................................Failed
- hook id: check-newsfragments
- exit code: 1

Traceback (most recent call last):
  File "C:\Program Files\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Or\Documents\Studies\TAU\Term2\Project2\Source_code\pylint_Forked\script\check_newsfragments.py", line 86, in <module>
    sys.exit(main())
  File "C:\Users\Or\Documents\Studies\TAU\Term2\Project2\Source_code\pylint_Forked\script\check_newsfragments.py", line 42, in main
    is_valid &= check_file(Path(filename), args.verbose)
  File "C:\Users\Or\Documents\Studies\TAU\Term2\Project2\Source_code\pylint_Forked\script\check_newsfragments.py", line 61, in check_file
    print(
  File "C:\Program Files\Python38\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode characters in position 76-77: character maps to <undefined>

```

### Expected behavior

Error message that ideally states which file is the problematic file, and at least say that the error is that there is no `.` in the end of fragment description, or that it does not match the template, etc..

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.12.10
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)]
```

### OS / Environment

Windows 10 Pro 64 bit

**Comment by Pierre-Sassoulas** [comment]
Is your branch up to date with the current main branch ? I think I did something for the ``.`` some time ago (very annoying to troubleshoot btw, sorry about that). Maybe it's only for the ``newsfragments``job ?

**Comment by orSolocate** [comment]
I just reproduced the error with a simpler case. Steps to reproduce updated in bug description above.
Hope you could reproduce it too on non-Windows machine..

**Comment by Pierre-Sassoulas** [comment]
It seems it's working well on pre-commit ci (https://results.pre-commit.ci/run/github/47671127/1665483920.oblgTTWySP--Hwi06zGMcw) but there is a crash on Windows if I understand right ? Sorry for that, we do not have a lot of active maintainers on Windows so it's almost never tested well.

**Comment by DudeNr33** [comment]
I have a windows machine, but I am running into my own problems with `pre-commit` right now (Can't find the Python interpreter when running through `pre-commit`).

@orSolocate can you please verify if you also experience this problem when directly running `python -m script.check_newsfragments doc\whatsnew\fragments\7777.new_check`?

The `UnicodeEncodeError` and the traceback looks like it is the emoijs in the `print` call that cause the trouble:
https://github.com/PyCQA/pylint/blob/c90029e78f7d6a0681199772933f20cd349d5847/script/check_newsfragments.py#L63

Windows does not use UTF-8, and AFAIK the encoding used can also differ on your system locale.
On my Windows machine with `cp1252` encoding for example the check runs fine and prints the emojis without problems.
The easiest solution might be to get rid of the emojis.

**Comment by Pierre-Sassoulas** [comment]
Let's remove the emojis then. I'm pretty sure I already had this problem before but sadly I don't remember in which project. black handle multi-platform emojis correctly and they seem to use ``echo`` from ``click``, but we don't have ``click`` (https://github.com/psf/black/blob/f16333e78ba77ab29ab0b75e10891e6c65c6da7e/src/black/output.py#L20). Those emojis are not nice enough to warrant more research 😄

**Comment by DanielNoord** [comment]
That was `pydocstringformatter`! You could check how we write to stdout there to see if that would help.

**Comment by DudeNr33** [comment]
Thanks for the hint, @DanielNoord!
https://github.com/DanielNoord/pydocstringformatter/blob/main/pydocstringformatter/_utils/output.py

@orSolocate as I can't reproduce it on my Windows machine, could you try replacing the `print` calls with the following?

```python
sys.stdout.buffer.write(string.encode("utf-8"))
```

... where `string` is the string which is currently passed into `print()`.

**Comment by Pierre-Sassoulas** [comment]
> That was pydocstringformatter!

I missed the pull request in the list when I searched ! Thank you @DanielNoord for saving the emojis 😄

**Comment by orSolocate** [comment]
@DudeNr33 I'd try and update

**Comment by orSolocate** [comment]
> python -m script.check_newsfragments doc\whatsnew\fragments\7777.new_check

When I run `python -m script.check_newsfragments doc\whatsnew\fragments\7777.new_check` I don't get the error.

But the emojis indeed are not recognized i my machine in the output message:

`does not respect the standard format ��`
